### PR TITLE
fix(visibility): Reset visibility

### PR DESCRIPTION
**Pull request by FlashHit**

### CODE REVIEW

File: ext/Client/__init__.lua [hunk 1]
```diff
@@ -578,43 +591,41 @@ function VEManagerClient:CreateEntity(p_Class, p_Guid)
...
 
 		-- t = elapsed time
 		-- b = begin
 		-- c = change == ending - beginning
 		-- d = duration (total time)
 		local t = TimeSinceStarted
-		local b = self.m_Presets[preset].startValue
-		local c = self.m_Presets[preset].EndValue - self.m_Presets[preset].startValue
-		local d = self.m_Presets[preset].time
+		local b = self.m_Presets[p_Id].startValue
```
Comment (IllustrisJack) [inline ext/Client/__init__.lua]: l_ID

File: ext/Client/__init__.lua [hunk 2]
```diff
@@ -118,6 +119,18 @@ function VEManagerClient:DisablePreset(p_ID)
 	self.m_Presets[p_ID].entity:FireEvent("Disable")
 end
 
+---@param p_ID string|nil
+---@param p_Visibility number|nil
+function VEManagerClient:ResetVisibility(p_ID, p_Visibility)
```
Comment (IllustrisJack) [inline ext/Client/__init__.lua]: SetVisibility()

File: ext/Client/__init__.lua [hunk 3]
```diff
@@ -118,6 +119,18 @@ function VEManagerClient:DisablePreset(p_ID)
...
+	for l_Index, l_ID in pairs(self.m_Lerping) do
+		if l_ID == p_ID then
+			self.m_Lerping[l_Index] = nil
+		end
+	end
+
+	self:SetVisibility(p_ID, p_Visibility)
```
Comment (IllustrisJack) [inline ext/Client/__init__.lua]: SetVisibilityInternal()

File: ext/Client/__init__.lua [hunk 4]
```diff
@@ -591,43 +591,41 @@ function VEManagerClient:CreateEntity(p_Class, p_Guid)
 end
 
 function VEManagerClient:UpdateLerp(percentage)
-	for i,preset in pairs(self.m_Lerping) do
+	for p_Id, _ in pairs(self.m_Lerping) do
```
Comment (IllustrisJack) [inline ext/Client/__init__.lua]: l_ID

File: ext/Client/__init__.lua [hunk 5]
```diff
@@ -591,43 +591,41 @@ function VEManagerClient:CreateEntity(p_Class, p_Guid)
...
 
-		local TimeSinceStarted = SharedUtils:GetTimeMS() - self.m_Presets[preset].startTime
-		local PercentageComplete = TimeSinceStarted / self.m_Presets[preset].time
+		local TimeSinceStarted = SharedUtils:GetTimeMS() - self.m_Presets[p_Id].startTime
```
Comment (IllustrisJack) [inline ext/Client/__init__.lua]: l_ID

File: ext/Client/__init__.lua [hunk 6]
```diff
@@ -591,43 +591,41 @@ function VEManagerClient:CreateEntity(p_Class, p_Guid)
...
+		local PercentageComplete = TimeSinceStarted / self.m_Presets[p_Id].time
```
Comment (IllustrisJack) [inline ext/Client/__init__.lua]: l_ID

File: ext/Client/__init__.lua [hunk 7]
```diff
@@ -591,43 +591,41 @@ function VEManagerClient:CreateEntity(p_Class, p_Guid)
...
 		-- t = elapsed time
 		-- b = begin
 		-- c = change == ending - beginning
 		-- d = duration (total time)
 		local t = TimeSinceStarted
-		local b = self.m_Presets[preset].startValue
-		local c = self.m_Presets[preset].EndValue - self.m_Presets[preset].startValue
-		local d = self.m_Presets[preset].time
+		local b = self.m_Presets[p_Id].startValue
+		local c = self.m_Presets[p_Id].EndValue - self.m_Presets[p_Id].startValue
```
Comment (IllustrisJack) [inline ext/Client/__init__.lua]: l_ID and so on

File: ext/Client/__init__.lua [hunk 8]
```diff
@@ -591,43 +591,41 @@ function VEManagerClient:CreateEntity(p_Class, p_Guid)
...
 		local lerpValue = m_Easing[transition](t,b,c,d)
 
 		if PercentageComplete >= 1 or PercentageComplete < 0 then
-			self:SetVisibility(preset, self.m_Presets[preset].EndValue)
-			self.m_Lerping[i] = nil
+			self:SetVisibility(p_Id, self.m_Presets[p_Id].EndValue)
+			self.m_Lerping[p_Id] = nil
 		else
-			self:SetVisibility(preset, lerpValue)
+			self:SetVisibility(p_Id, lerpValue)
```
Comment (IllustrisJack) [inline ext/Client/__init__.lua]: SetVisibilityInternal()

File: ext/Client/__init__.lua [hunk 9]
```diff
@@ -118,6 +119,18 @@ function VEManagerClient:DisablePreset(p_ID)
...
+			self.m_Lerping[l_Index] = nil
+		end
+	end
+
+	self:SetVisibility(p_ID, p_Visibility)
+end
+
 ---@param p_ID string|nil
 ---@param p_Visibility number|nil
 function VEManagerClient:SetVisibility(p_ID, p_Visibility)
```
Comment (IllustrisJack) [inline ext/Client/__init__.lua]: SetVisibilityInternal()

**Review by IllustrisJack — CHANGES_REQUESTED** [review]
this and merge

File: ext/Client/__init__.lua (hunk 8 above)
Comment (IllustrisJack) [reply ext/Client/__init__.lua]: l_ID

File: ext/Client/__init__.lua [hunk 10]
```diff
@@ -63,6 +63,7 @@ function VEManagerClient:RegisterEvents()
 	Events:Subscribe('VEManager:RegisterPreset', self, self.RegisterPreset)
 	Events:Subscribe('VEManager:EnablePreset', self, self.EnablePreset)
 	Events:Subscribe('VEManager:DisablePreset', self, self.DisablePreset)
+	Events:Subscribe('VEManager:ResetVisibility', self, self.ResetVisibility)
```
Comment (IllustrisJack) [inline ext/Client/__init__.lua]: obsolete after review changes

**Review by 3ti65 — APPROVED** [review]

**Pull request merged**